spring-integration-core/src/main/java/org/springframework/integration/config/ConsumerEndpointFactoryBean.java

```diff
@@ -210,6 +210,14 @@ public void afterPropertiesSet() {
 			}
 		}
 
+		adviceChain();
+		if (this.channelResolver == null) {
+			this.channelResolver = new BeanFactoryMessageChannelDestinationResolver(this.beanFactory);
```

Comment:
Why is this not a `BeanFactoryChannelResolver`?
Although why don't you use a `ChannelResolverUtils.getChannelResolver()` instead any way?

Reply:
Hehe - you tell me: https://github.com/spring-projects/spring-integration/pull/1359

Reply:
Oh! Looks like that one was an omission... 😄 

Or an idea was that this kind of `DestinationResolver` doesn't look into the `HeaderChannelRegistry` since we really don't have any headers interaction at this point...

Let's move it to the `ChannelResolverUtils.getChannelResolver()` anyway!

I don't think it is going to be some kind of an overhead that we still rely here on our global `BeanFactoryChannelResolver` bean

Reply:
Yikes, I pushed it to master instead of to this PR 😦

Reply:
OK. So, rebase then.
That's why I try to be more specific in every commit message.
`Polishing - PR comment` is not so informative 😉